service/history/api/respondworkflowtaskcompleted/workflow_task_completed_handler.go

```diff
@@ -372,6 +372,14 @@ func (handler *workflowTaskCompletedHandler) handleMessage(
 	case update.ProtocolV1:
 		upd := handler.updateRegistry.Find(ctx, message.ProtocolInstanceId)
 		if upd == nil {
+			upd, err = handler.updateRegistry.TryResurrect(ctx, message)
+			if err != nil {
+				return handler.failWorkflowTaskOnInvalidArgument(
+					enumspb.WORKFLOW_TASK_FAILED_CAUSE_BAD_UPDATE_WORKFLOW_EXECUTION_MESSAGE, err)
+			}
```

Comment:
`TryResurrect` is written assuming this is an Acceptance or Rejection protocol message. Are you sure it's impossible for `upd == nil` with a Completion protocol message? Maybe at least `TryResurrect` should handle that case and emit a helpful error message?